### ISSUE TITLE
PD Pod cannot be deleted

### ISSUE THREAD

**Issue by DanielZhangQD** [pingcap/tidb-operator#1567]
## Bug Report

**What version of Kubernetes are you using?**

v1.12.8
**What version of TiDB Operator are you using?**

tidb-operator:bfa1983847315df80b4c3f5d88f340de16018aa2
**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

local-storage
**What's the status of the TiDB cluster pods?**

Running
**What did you do?**

Scale in PD
**What did you expect to see?**
PD is scaled in successfully
**What did you see instead?**
PD is not scaled in

**Comment by DanielZhangQD** [comment]
Here is the log from kube-controller-manager:
```
E0116 02:12:38.461930       1 stateful_set.go:400] Error syncing StatefulSet tiflash-dt3/tiflash-dt-cluster3-pd, requeuing: admission webhook "delete.podadmission.tidb.pingcap.com" denied the request: persistentvolumeclaims "tikv-tiflash-dt-cluster3-pd-3" not found
I0116 02:12:38.462072       1 event.go:221] Event(v1.ObjectReference{Kind:"StatefulSet", Namespace:"tiflash-dt3", Name:"tiflash-dt-cluster3-pd", UID:"8022e2d1-3786-11ea-8799-eeeeeeeeeeee", APIVersion:"apps/v1", ResourceVersion:"392308714", FieldPath:""}): type: 'Warning' reason: 'FailedDelete' delete Pod tiflash-dt-cluster3-pd-3 in StatefulSet tiflash-dt-cluster3-pd failed error: admission webhook "delete.podadmission.tidb.pingcap.com" denied the request: persistentvolumeclaims "tikv-tiflash-dt-cluster3-pd-3" not found
```

Here is the log from admission controller:
```
I0116 02:02:38.387886       1 pods.go:88] receive DELETE pod[tiflash-dt3/tiflash-dt-cluster3-pd-3] by sa[system:serviceaccount:kube-system:statefulset-controller]
I0116 02:02:38.387916       1 pods.go:114] receive admission to delete pod[tiflash-dt3/tiflash-dt-cluster3-pd-3]
I0116 02:02:38.404238       1 pd_deleter.go:67] receive delete pd pod[tiflash-dt3/tiflash-dt-cluster3-pd-3] of tc[tiflash-dt3/tiflash-dt-cluster3],isMember=false,isInOrdinal=false,isUpgrading=false,isDeferDeleting=true,isLeader=false
I0116 02:03:08.681965       1 pods.go:88] receive DELETE pod[tiflash-dt3/tiflash-dt-cluster3-pd-3] by sa[system:serviceaccount:kube-system:statefulset-controller]
I0116 02:03:08.681990       1 pods.go:114] receive admission to delete pod[tiflash-dt3/tiflash-dt-cluster3-pd-3]
I0116 02:03:08.715838       1 pd_deleter.go:67] receive delete pd pod[tiflash-dt3/tiflash-dt-cluster3-pd-3] of tc[tiflash-dt3/tiflash-dt-cluster3],isMember=false,isInOrdinal=false,isUpgrading=false,isDeferDeleting=true,isLeader=false
I0116 02:03:33.116895       1 statefulset.go:64] admit {apps v1 statefulsets} [tiflash-dt3/tiflash-dt-cluster3-pd]
I0116 02:03:38.629053       1 pods.go:88] receive DELETE pod[tiflash-dt3/tiflash-dt-cluster3-pd-3] by sa[system:serviceaccount:kube-system:statefulset-controller]
I0116 02:03:38.629074       1 pods.go:114] receive admission to delete pod[tiflash-dt3/tiflash-dt-cluster3-pd-3]
I0116 02:03:38.673212       1 pd_deleter.go:67] receive delete pd pod[tiflash-dt3/tiflash-dt-cluster3-pd-3] of tc[tiflash-dt3/tiflash-dt-cluster3],isMember=false,isInOrdinal=false,isUpgrading=false,isDeferDeleting=true,isLeader=false
```

**Comment by DanielZhangQD** [comment]
1. Code [here](https://github.com/pingcap/tidb-operator/blob/master/pkg/webhook/pod/pd_deleter.go#L124) should be `PDMemberType`, right?
2. If PVC does not exist, I think we should permit the deletion.

**Comment by Yisaer** [comment]
It seems the admission-webhook can't found the pvc of the pod so that it can't annotate the pvc and return the error.
If PVC does not exist, the request could be permitted.

**Comment by DanielZhangQD** [comment]
The MemberType used in the code to get PVC is incorrect, that's why webhook cannot get the PVC.

**Comment by Yisaer** [comment]
Fixed in #1568
